### DISCUSSION TITLE
mac升级了clang后编译luamake报错

### DISCUSSION THREAD

**Discussion by fdgggy** [ejoy/ant#84] [Q&A]
[1/41] Compile C   build/macos/debug/obj/zlib-ng-arm/crc32_acle.obj
FAILED: build/macos/debug/obj/zlib-ng-arm/crc32_acle.obj
clang -MMD -MT build/macos/debug/obj/zlib-ng-arm/crc32_acle.obj -MF build/macos/debug/obj/zlib-ng-arm/crc32_acle.obj.d -std=c17  -Wall -g -I./3rd/zlib-ng -I./3rd/zlib-ng/arch/arm/ -Ibuild/macos/debug/gen-zlib -DARM_FEATURES -DARM_NEON -DARM_NEON_HASLD4 -DARM_ACLE -DHAVE_THREAD_LOCAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_BUILTIN_CTZLL -o build/macos/debug/obj/zlib-ng-arm/crc32_acle.obj -c 3rd/zlib-ng/arch/arm/crc32_acle.c
In file included from 3rd/zlib-ng/arch/arm/crc32_acle.c:12:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/arm_acle.h:14:2: error: "ACLE intrinsics support not enabled."
#error "ACLE intrinsics support not enabled."
 ^
3rd/zlib-ng/arch/arm/crc32_acle.c:23:13: error: call to undeclared function '__crc32b'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        c = __crc32b(c, *buf++);
            ^
3rd/zlib-ng/arch/arm/crc32_acle.c:29:13: error: call to undeclared function '__crc32h'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        c = __crc32h(c, *buf2++);
            ^
3rd/zlib-ng/arch/arm/crc32_acle.c:77:13: error: call to undeclared function '__crc32w'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        c = __crc32w(c, *buf4++);
            ^
3rd/zlib-ng/arch/arm/crc32_acle.c:83:13: error: call to undeclared function '__crc32h'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        c = __crc32h(c, *buf2++);
            ^
3rd/zlib-ng/arch/arm/crc32_acle.c:92:13: error: call to undeclared function '__crc32b'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        c = __crc32b(c, *buf);
            ^

**Comment by fdgggy** [comment]
放弃了，折腾了几个小时没搞成，大家用mac的解决了的可以贴出来一哈。

**Comment by fdgggy** [comment]
windows成功编译，非常顺利，mac的优先级应该比较低，为节省时间，还是建议大家直接用windows

**Reply by julywind168** [reply]
😑 mac 一直可以编译, 但编辑器问题比较多

**Comment by Gowa2017** [comment]
他这个编译是有问题的，默认情况下，他将 macos 依赖的是 zlib arm 很遗憾，我使用的是x86平台，所以得自己改变异的luamake 。但是这是啥东西？
为什么会有个Luamake 出来，学习成本太高了。

**Reply by fdgggy** [reply]
先用windows把，开发组都是windows，环境问题的优先级不高。

**Comment by Gowa2017** [comment]
编译出来没有问题了 看我给的pr 但是luamake么办法获取版本号是个问题
